ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java

```diff
@@ -314,21 +324,59 @@ public OutputStream _invoke(final String opName, final InputStream in, final Res
                                 retVal = false;
                             }
                         } else {
-
-                            if (sc != null) {
-                                setSecurityContextOnAssociation(sc);
+                            if (this.securityDomain != null) {
+                                // an elytron security domain is available: authenticate and authorize the client before
+                                // invoking the component.
+                                ServerAuthenticationContext context = this.securityDomain.createNewAuthenticationContext();
+                                if (credential == sasCurrent) {
+                                    // TODO change this to use the SASCurrent as the credential when we have a realm that can act on it.
```

Comment:
I don't think we need to use the SASCurrent as a credential: I bet we could just take get_incoming_password(), decode that to characters, and then use that as a ClearPassword.